src/PhpSpreadsheet/Calculation/DateTime.php

```diff
@@ -1301,31 +1313,34 @@ public static function WEEKNUM($dateValue = 1, $method = 1)
 
         if (!is_numeric($method)) {
             return Functions::VALUE();
-        } elseif (($method < 1) || ($method > 2)) {
-            return Functions::NAN();
         }
-        $method = floor($method);
+        $method = (int) $method;
+        $methodarr = [1 => 1, 2, 11 => 2, 3, 4, 5, 6, 7, 1, 21 => 21];
```

Comment:
Would this array be better as a static property, or even a constant array; and perhaps with name constants for the keys to provide a human-readable explanation of the $method names that can then be use in the body of the code?

Reply:
Recoded array as constant (didn't even know that was possible), plus other named constants per this review comment.